lib/auth/join_kubernetes.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/authentication/v1"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/kubernetestoken"
+)
+
+type kubernetesTokenValidator interface {
+	Validate(context.Context, string) (*v1.UserInfo, error)
+}
+
+func (a *Server) checkKubernetesJoinRequest(ctx context.Context, req *types.RegisterUsingTokenRequest) error {
+	if req.IDToken == "" {
+		return trace.BadParameter("IDToken not provided for Kubernetes join request")
+	}
+	pt, err := a.GetToken(ctx, req.Token)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	userInfo, err := a.kubernetesTokenValidator.Validate(ctx, req.IDToken)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	log.WithFields(logrus.Fields{
+		"userInfo": userInfo,
+		"token":    pt.GetName(),
+	}).Info("Kubernetes workload trying to join cluster")
+
+	// TODO: if pod extradata compare with node name?
```

Comment:
Does this TODO need addressing now ? If it's for a later date, can we include a user who is responsible for this TODO (e.g `TODO(hugo)`)

Reply:
It should be removed

```suggestion
```